### PR TITLE
Mysql database multiple sql statements

### LINKED ISSUE #151: mysql_database with action :query cannot execute several sql statements
Let's assume we have: 

```
mysql_database 'default' do
  database_name 'default'
  connection mysql_connection_info
  sql <<-EOH
CREATE TABLE IF NOT EXISTS table1 (
  client_id VARCHAR(80) NOT NULL,
);
CREATE TABLE IF NOT EXISTS table2 (
  client_id VARCHAR(80) NOT NULL,
);
CREATE TABLE IF NOT EXISTS table3 (
  client_id VARCHAR(80) NOT NULL,
);
EOH
  action :query
end
```

And as MySQL2 gem: 

```
mysql2_chef_gem 'default' do
  gem_version '0.3.17'
  action :install
end
```

When running chef-client we get and error similar to this:

```
ERROR: mysql_database[default] (my cookbook::bootstrap_db line xx) had an error: Mysql2::Error: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'CREATE TABLE IF NOT EXISTS table2 (
 client_id VARCHAR(80) NOT NULL,' at line xx
```

Table2 is the hint. 

After some debugging we found that the "guilty" is the MySql2 gem that cannot execute several sql statements unless you specify:

```
Mysql2::Client.default_query_options[:connect_flags] |= Mysql2::Client::MULTI_STATEMENTS
```

as read in:  http://stackoverflow.com/questions/4856759/how-to-run-multiple-sql-queries-through-one-call-in-mysql2-gem-in-irb

for example in a ruby block:

```
ruby_block "modifying MySQL2 client" do
  block do
    Mysql2::Client.default_query_options[:connect_flags] |= Mysql2::Client::MULTI_STATEMENTS
  end
  action :run
end
```

After adding that, chef-client completes without any complain. 

After looking at the provider I think the query options must be somewhere near here:
database/libraries/provider_database_mysql.rb :138

```
def test_client
  require 'mysql2'
  @test_client ||=
      Mysql2::Client.new(
          host: new_resource.connection[:host],
          socket: new_resource.connection[:socket],
          username: new_resource.connection[:username],
          password: new_resource.connection[:password],
          port: new_resource.connection[:port]
      )
end
```

I never made a pull request so I wanted to share this before doing it.

Where it should be the best place to specify it? or if this was left out on purpose by design and instead we should use different approach. 

Thanks,

### PR DESCRIPTION
**Pull request by razor-x**
@someara I've created this pull request by rebasing @fonzai's [mysql_database-multiple-sql-statements](https://github.com/fonzai/database/tree/) branch. This is to fix #151.

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @fonzai

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by fonzai** [comment]
Thanks @razor-x! So did I get it right, I have to sign the CLA before you can merge it?

**Comment by razor-x** [comment]
@fonzai Using your branch solved this issue for me, so I figured I would just get the ball rolling with a pull request and see what the maintainers think about merging it. I think you do need to sign the CLA for anything to move forward though. Even though this is from my fork, the commits are still rightflly under your authorship.

If prefer to submit a pull request from your fork so you can continue to edit this patch directly, let me know and I'll close this one.

**Comment by fonzai** [comment]
@razor-x Sounds good, hanks for help! I will handle the CLA part after I have talked my employer how they want it done (I see there are two types of the CLA available).

File: libraries/provider_database_mysql.rb [hunk 1]
```diff
@@ -136,22 +120,27 @@ def close_repair_client
...
+          @query_client.close
+        rescue Mysql2::Error
+          @query_client = nil
+        end
+
+        def create_mysql_query_client
           require 'mysql2'
-          @query_client ||=
-            Mysql2::Client.new(
+          Mysql2::Client.default_query_options[:connect_flags] |= Mysql2::Client::MULTI_STATEMENTS
```
Comment (thorstenk) [inline libraries/provider_database_mysql.rb]: This changes the connection flags for *all* connections.
See my comments to 5928ecc0df11d7c50

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @fonzai

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by tas50** [comment]
At this time I'm going to close this issue out as we've deprecated this cookbook. The original intent was to abstract database implementation details away from the end user. In hindsight this was a mistake as it is difficult / impossible to make a generic interface for databases that works across all systems. Instead we plan to move the individual providers into their respective cookbooks (mysql, postgresql, sqlite, sql_server) where they can fully utilize the feature set of the underlying DB systems.

**Pull request closed without merging**